test/csuite/wt7989_compact_checkpoint/main.c

```diff
@@ -0,0 +1,286 @@
+/*-
+ * Public Domain 2014-present MongoDB, Inc.
+ * Public Domain 2008-2014 WiredTiger, Inc.
+ *
+ * This is free and unencumbered software released into the public domain.
+ *
+ * Anyone is free to copy, modify, publish, use, compile, sell, or
+ * distribute this software, either in source code form or as a compiled
+ * binary, for any purpose, commercial or non-commercial, and by any
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+/*
+ * This test executes to test cases: 
+ * - One with WT_TIMING_STRESS_CHECKPOINT_SLOW flag. It adds 10 secods sleep before each checkpoint.
+ * - Another test case synchronizes compact and checkpoint threads using a condition variable.
+ * The reason we have two test here is that they give different output when configured
+ * with "verbose=[compact,compact_progress]". There's a chance these two cases are different.
+ */
+
+#define NUM_RECORDS 1000000
+
+/* Constants and variables declaration. */
+/* 
+ * You may want to add "verbose=[compact,compact_progress]" to the connection config stringto get
+ * better view on what is happening.
+ */
+static const char conn_config[] =
+  "create,cache_size=2GB,statistics=(all)";
+static const char table_config[] =
+  "allocation_size=4KB,leaf_page_max=4KB,key_format=i,value_format=QQQS";
+static char data_str[1024] = "";
+
+/* Structures definition. */
+struct thread_data {
+    WT_CONNECTION *conn;
+    const char *uri;
+    WT_CONDVAR *cond;
+    WT_SESSION_IMPL *session_cond;
+};
+
+/* Forward declarations. */
+static void run_test(const char *home, const char *uri, bool stress_test);
```

Comment:
You may want to alphabetize that part, just a suggestion though.

Reply:
Done